lib/facility.rb

```diff
@@ -1,14 +1,107 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services,
+              :registered_vehicles,
+              :collected_fees
+  
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_info)
+    @name = facility_info[:name]
+    @address = facility_info[:address]
+    @phone = facility_info[:phone]
     @services = []
+    @collected_fees = 0
+    @registered_vehicles = []
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+
+  def register_vehicle(automobile)
+    if @services.include?('Vehicle Registration')
+    automobile.registration_date = Date.today
+    @registered_vehicles << automobile
+      if automobile.antique? == true
```

Comment:
setting the plate type and collecting fees here can add a good amount of complexity, making this method pretty big. Consider creating one or two helper methods that you can call here to handle the plate type and fee collection. For example, your method could become:
```
 def register_vehicle(automobile)
    if @services.include?('Vehicle Registration')
        automobile.registration_date = Date.today
        @registered_vehicles << automobile
        collect_fees(automobile)
        automobile.set_plate_type
        return @registered_vehicles
    else
        return nil
   end
end
```

Then, `collect_fees` could be another method in this class that determines the fees to add to @collected_fees based on the car type. `set_plate_type` could be a method in the Vehicle class to change the plate_type since thats more of a vehicle responsibility than something the facility needs to worry about.